### PR TITLE
Add documentation for textworld.gym

### PR DESCRIPTION
**Pull request by MarcCote**
This PR adds documentation for the textworld.gym API.

### CODE REVIEW

File: textworld/gym/core.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from typing import List, Mapping, Any, Optional
+
+from textworld import EnvInfos
+
+
+class Agent:
+    """ Interface for any agent playing TextWorld games. """
+
+    @property
+    def infos_to_request(self) -> EnvInfos:
```
Comment (tavianator) [inline textworld/gym/core.py]: This doesn't match the current contest API (`select_additional_infos()`), I guess that's okay?

Comment (MarcCote) [reply textworld/gym/core.py]: Yeah, I didn't like `select_additional_infos` so I'm changing it now :P. That said, this abstract agent class is not used yet.

**Comment by tavianator** [comment]
AFAICT this was already included in #124

File: textworld/gym/core.py (hunk 1 above)
Comment (tavianator) [reply textworld/gym/core.py]: Yeah sounds good.  Like you say, there's no framework users yet.  We can always make the contest compatible with both agent APIs.

**Comment by MarcCote** [comment]
Yes, #124 was based on top of #125 and #126.

**Pull request closed without merging**